tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -86,7 +86,7 @@ protected void recycle(T element) {
   private BlockingQueue<T> elements;
 
   // number of current acvtive elements (including ones that are in use)
-  int size = 0;
+  volatile int size = 0;
```

Comment:
nice catch. This must have never caused trouble because when we use in CDAP, we know the number of threads and configure a large enough pool.